From b9e4569769f8ce26991e72151ac33e216dbd40ff Mon Sep 17 00:00:00 2001
From: MouriNaruto <Kenji.Mouri@outlook.com>
Date: Thu, 11 Dec 2025 11:36:01 +0800
Subject: [PATCH] Enable Mile.Windows.UniCrt integration support.

---
 cmake/QtBuildHelpers.cmake | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/cmake/QtBuildHelpers.cmake b/cmake/QtBuildHelpers.cmake
index af7070e232b..3cdd0547e9a 100644
--- a/cmake/QtBuildHelpers.cmake
+++ b/cmake/QtBuildHelpers.cmake
@@ -491,4 +491,6 @@ macro(qt_internal_setup_build_and_global_variables)
     qt_internal_set_qt_allow_download()
 
     qt_internal_detect_dirty_features()
+
+    include(${CMAKE_CURRENT_LIST_DIR}/../../Mile.Windows.UniCrt/Mile.Windows.UniCrt.cmake)
 endmacro()
-- 
2.51.2.windows.1

