From aa85b050fe673869920382aa015ed66b99cfa51d Mon Sep 17 00:00:00 2001
From: Kenji Mouri <kenji.mouri@outlook.com>
Date: Thu, 04 Dec 2025 20:00:48 +0800
Subject: [PATCH] QWindowsFontDatabase: Fix build issues when disabling DirectWrite

Pick-to: 6.11
Change-Id: I0b2bf2983edb2f95b134f6812dab05bc633b52fc
Reviewed-by: Zhao Yuhang <2546789017@qq.com>
Reviewed-by: Joerg Bornemann <joerg.bornemann@qt.io>
---

diff --git a/src/gui/text/windows/qwindowsfontdatabase.cpp b/src/gui/text/windows/qwindowsfontdatabase.cpp
index 93af1a9..0c6d9a3 100644
--- a/src/gui/text/windows/qwindowsfontdatabase.cpp
+++ b/src/gui/text/windows/qwindowsfontdatabase.cpp
@@ -1113,17 +1113,21 @@
     m_eudcFonts.clear();
 }
 
+#if QT_CONFIG(directwrite)
 QWindowsFontDatabase::FontHandle::FontHandle(IDWriteFontFace *face, const QString &name)
     : fontFace(face), faceName(name)
 {
     fontFace->AddRef();
 }
+#endif // !QT_NO_DIRECTWRITE
 
 
 QWindowsFontDatabase::FontHandle::~FontHandle()
 {
+#if QT_CONFIG(directwrite)
     if (fontFace != nullptr)
         fontFace->Release();
+#endif // !QT_NO_DIRECTWRITE
 }
 
 void QWindowsFontDatabase::releaseHandle(void *handle)
diff --git a/src/gui/text/windows/qwindowsfontdatabase_p.h b/src/gui/text/windows/qwindowsfontdatabase_p.h
index 92e3f04..856a559 100644
--- a/src/gui/text/windows/qwindowsfontdatabase_p.h
+++ b/src/gui/text/windows/qwindowsfontdatabase_p.h
@@ -78,10 +78,14 @@
 
     struct FontHandle {
         FontHandle(const QString &name) : faceName(name) {}
+#if QT_CONFIG(directwrite)
         FontHandle(IDWriteFontFace *face, const QString &name);
+#endif // !QT_NO_DIRECTWRITE
         ~FontHandle();
 
+#if QT_CONFIG(directwrite)
         IDWriteFontFace *fontFace = nullptr;
+#endif // !QT_NO_DIRECTWRITE
         QString faceName;
     };
 
